src/Workspaces/Core/Portable/Workspace/Solution/Solution.cs

```diff
@@ -827,7 +827,16 @@ public Solution WithProjectParseOptions(ProjectId projectId, ParseOptions option
                 return this;
             }
 
-            return this.ForkProject(newProject, CompilationTranslationAction.ProjectParseOptions(newProject));
+            if (this.Workspace.PartialSemanticsEnabled)
+            {
+                // don't fork tracker with queued action since access via partial semantics can become inconsistent (throw).
+                // Since changing options is rare event, it is okay to start compilation building from scratch.
+                return this.ForkProject(newProject, forkTracker: false);
```

Comment:
Why is this conditional on PartialSemanticsEnabled? It seems even if it's not, there's no benefit to forking the tracker -- it'll still have to rebuild everything anyways so why hold onto the old state?